packages/vue-client/src/components/GameTimer.vue

```diff
@@ -0,0 +1,93 @@
+<script setup lang="ts">
+import { isDefined } from "@vueuse/core";
+import { onMounted, onUnmounted, ref, watch } from "vue";
+import type { IPerPlayerTimeControlBase } from "@ogfcommunity/variants-shared";
+
+const props = defineProps<{
+  time_control: IPerPlayerTimeControlBase | null;
+}>();
+
+const time = ref(props.time_control?.remainingTimeMS ?? 0);
+const formattedTime = ref(
+  props.time_control?.remainingTimeMS
+    ? msToTime(props.time_control?.remainingTimeMS)
+    : ""
+);
+const isCountingDown = ref(false);
+let timerIndex: number | null = null;
+
+watch(time, (t: number) => {
+  if (t === null) {
+    formattedTime.value = "";
+  } else {
+    formattedTime.value = msToTime(t);
+  }
+});
+
+onMounted(() => resetTimer());
+watch(props, () => resetTimer());
```

Comment:
I couple things about `watch`:

- watch has an `immediate` option that causes the function to run when the component is created, so that might be able to replace the `onMounted` call. ([docs](https://vuejs.org/guide/essentials/watchers.html#eager-watchers))
- The third argument of the callback is a `cleanup` function, so we could consolidate the `clearInterval` logic and remove the `onUnmounted` call. ([docs](https://vuejs.org/api/reactivity-core.html#watch))

```
watch(props, (value, oldValue, onCleanup) => {
      ...
    const timerIndex = setInterval(...);
      ...
    onCleanup(() => clearInterval(timerIndex));
}, {immediate: true})
```

Reply:
Thank you for your insights! I've changed it accordingly.